src/main/java/org/zalando/nakadi/domain/StrictJsonParser.java

```diff
@@ -0,0 +1,287 @@
+package org.zalando.nakadi.domain;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class StrictJsonParser {
```

Comment:
Can you please describe which part of this file was written by you and which was borrowed? So that I know where to focus the review.

Reply:
I took readStringTillTheEnd method completely. Everything else is more or less rewritten